### ISSUE TITLE
New SC Committee Scraper

### ISSUE THREAD

**Issue by chrisyamas** [openstates/issues#889]
### We need a new committee scraper written in [spatula](https://jamesturk.github.io/spatula/) for South Carolina.

*Special note about SC joint committees PDF page: scraper should a text content check (or other approach) built into it to verify if the PDF was published during current session)*

It should scrape this [webpage of Senate Committees](https://www.scstatehouse.gov/committee.php?chamber=S), this [webpage of House Committees](https://www.scstatehouse.gov/committee.php?chamber=H), and this [PDF webpage of Joint Committees](https://www.scstatehouse.gov/publicationspage/JtSpecCommList.pdf) to get:
- **name**
   - “Small Business”, “Agriculture”, etc.
- **chamber**
   - “upper”, “lower”, or “legislature” (i.e. joint/special)
- **classification**
   - ex: "committee", "subcommittee"
- **parent**
   - *only if it is a subcommittee* - scraper should have a way to determine this
   - ex: parent="Natural Resources" when *sub*committee name="Forestry"
- **sources**
   - each added using `add_source()` method on instance of `ScrapeCommittee` type object
   - ex: home page for list of committees, specific page for that committee, etc.
- **members**
   - - each added using `add_member()`method on instance of `ScrapeCommittee` type object
   - *Attributes:*
      - **name**: “Jane Doe”, “John Smith”, etc.
      - **role** (where applicable): “Chair”, “Ranking Member”, etc.

### Useful scrapers for reference
A [spatula-based scraper](https://github.com/openstates/openstates-scrapers/blob/main/scrapers_next/mo/committees.py) that also scrapes an HTML List Page (using `HtmlListPage` and `HtmlPage` spatula classes) for reference.
- As in the other spatula scraper, you will need to set the session variable separately (rather than depend on the `__init__.py`).

### Other useful resources
You can reference the [object models for committee scrapers](https://github.com/openstates/openstates-core/blob/main/openstates/models/committees.py) in the Open States Core repo.

Further documentation on running spatula scrapers in the command line can be found [here](https://docs.openstates.org/contributing/scrapers/#running-spatula-scrapers).

**Comment by fvescia** [comment]
Working on this!

**Comment by sroomf** [comment]
hi @fvescia, thanks for tackling this scraper! We would love it if you could finish up your work on this by **Monday, 1/31**. It's not a problem if you can't finish it up, but it is important for you to [push the work that you've done](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) so we can take over. Let us know if you have any questions or issues! We're happy to help out :)

**Comment by fvescia** [comment]
Hi @sroomf, sorry for the slow reply! I've only put a small dent in this – it's been a busy few weeks at school – but I'm happy to share what I have. The link in your comment is for pull request instructions, so I just tried the usual git add / git committ / git push, and it looks like I don't have permission to share my work that way (see below). Are there alternative steps I can take to push? Thanks!

```
@fvescia ➜ /workspaces/openstates-scrapers/scrapers_next/sc (main) $ git add committees.py 
@fvescia ➜ /workspaces/openstates-scrapers/scrapers_next/sc (main) $ git commit -m "Made a (small) start"
[main ee4ace4b0] Made a (small) start
 1 file changed, 35 insertions(+)
 create mode 100644 scrapers_next/sc/committees.py
@fvescia ➜ /workspaces/openstates-scrapers/scrapers_next/sc (main) $ git push
remote: Permission to openstates/openstates-scrapers.git denied to fvescia.
fatal: unable to access 'https://github.com/openstates/openstates-scrapers/': The requested URL returned error: 403
```

**Comment by fvescia** [comment]
Also, I'd be glad to keep working on this over the next week or two - let me know if that would be helpful, or if someone on the Open States team would rather take over to get it done ASAP

**Comment by sroomf** [comment]
Have you tried running `git push --set-upstream origin main`? Alternatively, if you haven't [set a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token), then that might be the issue. Happy to hop on a zoom call and figure it out together as well!


> Hi @sroomf, sorry for the slow reply! I've only put a small dent in this – it's been a busy few weeks at school – but I'm happy to share what I have. The link in your comment is for pull request instructions, so I just tried the usual git add / git committ / git push, and it looks like I don't have permission to share my work that way (see below). Are there alternative steps I can take to push? Thanks!
> 
> ```
> @fvescia ➜ /workspaces/openstates-scrapers/scrapers_next/sc (main) $ git add committees.py 
> @fvescia ➜ /workspaces/openstates-scrapers/scrapers_next/sc (main) $ git commit -m "Made a (small) start"
> [main ee4ace4b0] Made a (small) start
>  1 file changed, 35 insertions(+)
>  create mode 100644 scrapers_next/sc/committees.py
> @fvescia ➜ /workspaces/openstates-scrapers/scrapers_next/sc (main) $ git push
> remote: Permission to openstates/openstates-scrapers.git denied to fvescia.
> fatal: unable to access 'https://github.com/openstates/openstates-scrapers/': The requested URL returned error: 403
> ```

**Comment by sroomf** [comment]
> Also, I'd be glad to keep working on this over the next week or two - let me know if that would be helpful, or if someone on the Open States team would rather take over to get it done ASAP

Another week should be ok on our end! I'll check in with you around this time next week :)

**Comment by fvescia** [comment]
Hi @sroomf, I took this to @jamesturk's office hours this week, and he thinks it's a bug in the codespace that he'll need to look into. How should I share what I've done so far in the meantime? Should I attach the .py file here, or would another way be better? Thanks!

**Comment by chrisyamas** [comment]
Hi @fvescia! I'm sorry that you've been dealing with that codespace bug; that's frustrating. However, yes, attaching the file to this ticket here works for us! We'll grab it and build on the foundation you've provided. Thank you so much for taking the time contribute and make sure your contribution is shared!

**Comment by fvescia** [comment]
Here you go! It looks like GitHub doesn't support .py files as issue attachments, so I copy/pasted everything into a Word Doc. It's a very modest start – I'm new to scraping (read: slow at writing scrapers!) and my plate has been pretty full with schoolwork – but hopefully there's something here someone can run with :)
[committees.docx](https://github.com/openstates/issues/files/10610169/committees.docx)

**Comment by chrisyamas** [comment]
Thank you, @fvescia! I went ahead and turned your work into a draft pull request of a new committee file for the SC jurisdiction. I completely understand and relate to the hectic busyness of balancing schoolwork with outside projects. However, next time you want to get some open source contribution experience with our project, don't hesitate to reach out on GitHub or even email me directly at christopher@civiceagle.com, and we'd be happy to find a way for you to contribute and get some more scraper writing/editing or even code documentation experience!

**Comment by jamesturk** [comment]
Hi,

Sorry we didn't get the Codespace thing resolved in time.  I am still trying to figure out the intended workflow & actually have a call on Wednesday with someone from GitHub to learn a bit more and talk through the difficulties we encountered.

It seems like when the codespace is created it is created without a fork the person can push to, which was unexpected, and furthermore it doesn't have an SSH key already configured (which I figured would be the back up option).

While we were in the room a few people made it through via a dance where after a failed push it gave them another option, but I'm not sure why that was necessary or how to replicate it 100%.

The experience up until that point was much better than not having the codespaces, so I'm hopeful with a bit more configuration we can get things to a place where it makes it easier to contribute to the project & not hit a brick wall at the very end of the process.
